Aws networks

DaMandal0rian opened the pull request:
This PR makes major changes for network primitives and testnet networks:
- add gemini-3e
- add gemini-3f
- changes to devnet
- minor changes old gemini-3d
- changes to network primitives bootstrap nodes and nodes with new flags
-  change DSN port from 50000 to 30533
- fix pruning of containers
- fix volume mounting for farmer
- block rpc node ingress by removing `port` parameter in docker compose
- fix networking aws declarations
- add pruning parameters to all nodes except rpc
- remove `bootnodes` and add `dsn-reserved-peers`
-  change to compute optimized `m6a` aws instances (for gemini-3f)
- add domains bootstrap node
- added changes related to external address detection (https://github.com/subspace/subspace/pull/1753/files)
- notable changes (https://github.com/subspace/subspace/pull/1755, https://github.com/subspace/subspace/pull/1765, https://github.com/subspace/subspace/pull/1778)

closes #122

DaMandal0rian committed:
network changes

- update devnet and gemini-3d snapshots
- remove system domain
- changes to node to be compatible with upstream changes https://github.com/paritytech/substrate/pull/13384

DaMandal0rian committed:
Prepare Gemini-3e

- set gemini-3e configuration
- fix typo in path
- include upstream RPC changes

DaMandal0rian committed:
Update devnet snapshots

- update latest genesis hash devnet

DaMandal0rian committed:
network primitives changes

DaMandal0rian committed:
changes to machines and cidr blocks

- minor fix outputs gemini-3e
- vpc names

DaMandal0rian committed:
update gitignore

DaMandal0rian committed:
minor fix full node

DaMandal0rian committed:
Fixes to provisioners 

- use absolute paths to fix scp bug with copying
- set permissions and ownership to install directory since scp assumes root user and causes permission denied 
- change prune script
- disable domain node
- update docker compose flags for bootstrap config

DaMandal0rian committed:
Minor changes

- Change lifecycle
- update snapshot
- add rpc output

DaMandal0rian committed:
Change DSN Port

DaMandal0rian committed:
Changes for DSN and substrate nodes

DaMandal0rian committed:
Gemini-3e config launch

DaMandal0rian committed:
Fixes primitives

- remove ingress 50000
- fix escaping domain node sript

DaMandal0rian committed:
Configure volume mounting for farmer

DaMandal0rian committed:
reset flags

DaMandal0rian committed:
Handle pruning of docker resources better

DaMandal0rian committed:
changes for network primitives + new devnet

DaMandal0rian committed:
Fixes to subspace and substrate node parameters

DaMandal0rian commented on aws/network-primitives/scripts/create_bootstrap_node_compose_file.sh:
```diff
@@ -26,14 +30,34 @@ services:
+    echo "      - \"--bootstrap-nodes\"" >> ~/subspace/docker-compose.yml
+    echo "      - \"${dsn_addr}\"" >> ~/subspace/docker-compose.yml
+  fi
+done
```
@shamil-gadelshin Please check config above

DaMandal0rian commented on aws/network-primitives/scripts/create_full_node_compose_file.sh:
```diff
@@ -57,26 +58,18 @@ dsn_bootstrap_node_count=${4}
   echo "      \"--reserved-nodes\", \"${addr}\"," >> ~/subspace/docker-compose.yml
   echo "      \"--bootnodes\", \"${addr}\"," >> ~/subspace/docker-compose.yml
 done
 
```
Please check the changes here too @shamil-gadelshin

DaMandal0rian reviewed (commented):
@shamil-gadelshin changes here to check please!

DaMandal0rian committed:
Fix farmer DSN and domain node id

- set domain node id to evm for now

DaMandal0rian committed:
remove unused domain labels and ids

- remove system and payments
- change evm from id 3 to 0

DaMandal0rian committed:
add new flags and make interpolation uniform

- use fix ports for host and container and not env variables.
- make external IP detectable
- use latest devnet snapshot

DaMandal0rian committed:
add gemini-3f network

- minor (update devnet snapshot to gemini-2f pre-release)

DaMandal0rian committed:
Merge branch 'gemini-3f-prep' into aws-networks

vedhavyas commented on aws/devnet/main.tf:
```diff
@@ -2,17 +2,18 @@ module "devnet" {
+    docker-tag         = "snapshot-2023-aug-15"
     reserved-only      = false
     prune              = false
-    genesis-hash       = "8a94e770a50298516444025b8c075034f8be7745812358d25189175fe504fc66"
```
do we not need this anymore ?

DaMandal0rian replied:
it's currently set to empty string. Since we gather that value after launching the network on first run.

vedhavyas commented on aws/gemini-3e/main.tf:
```diff
@@ -0,0 +1,96 @@
+    instance-type      = var.instance_type
+    deployment-version = 1
+    regions            = var.aws_region
+    instance-count     = var.instance_count + 1
```
why +1 here though ?

DaMandal0rian replied:
instance count is `1` by default, it's increased to launch 2 nodes.

vedhavyas replied:
can't we just increase the actual value of instance_count in the vars instead of doing an addition here ?

DaMandal0rian replied:
sure, but then we will have 2 full nodes and 2 farmers and 2 domain nodes and eventually 2 domain bootstrap nodes. However i will change this to a map.

vedhavyas commented on aws/network-primitives/instances.tf:
```diff
@@ -2,8 +2,8 @@ resource "aws_instance" "bootstrap_node" {
-  subnet_id         = element(aws_subnet.public_subnets.*.id, count.index)
-  availability_zone = element(var.azs, count.index)
+  subnet_id         = element(aws_subnet.public_subnets.*.id, 0)
+  availability_zone = element(var.azs, 0)
```
does Az has to be the first one always ?
if so, why not just make it a single value instead of list ?

DaMandal0rian replied:
Yeah can make it a fixed string since we use one AZ now.

vedhavyas commented on aws/network-primitives/scripts/create_bootstrap_node_compose_file.sh:
```diff
@@ -21,58 +27,73 @@ services:
+      - \${DSN_NODE_KEY}
+      - /ip4/0.0.0.0/tcp/30533
       - --protocol-version
       - \${GENESIS_HASH}
```
Shouldn't the genesis hash be here  since i remember we are passing an empty value above ?

DaMandal0rian replied:
We pass an empty value first, and then attain it when launching the node and getting the `protocol_version` value. Once launched it's populated in `main.tf`.

vedhavyas commented on aws/network-primitives/scripts/create_domain_node_compose_file.sh:
```diff
@@ -82,60 +76,39 @@ dsn_bootstrap_node_count=${4}
     # core domain
       echo '      "--",'
       echo '      "--chain=${NETWORK_NAME}",'
+    #  echo '      "--alice",' //todo: make configurable with devnet
```
we can remove this. We will never have --alice on any networks ever

DaMandal0rian replied:
Great, will remove!

vedhavyas commented on aws/network-primitives/scripts/create_domain_node_compose_file.sh:
```diff
@@ -82,60 +76,39 @@ dsn_bootstrap_node_count=${4}
       echo '      "--chain=${NETWORK_NAME}",'
+    #  echo '      "--alice",' //todo: make configurable with devnet
       echo '      "--validator",'
+    #  echo '      "--enable-subspace-block-relay",'
```
would have preferred if this was optional rather than hand-made comment

DaMandal0rian replied:
This was just to disable since it wasn't working at the time and will be re-enabled in future once it's fixed (which it is from Rahul's recent testing).

vedhavyas commented on aws/network-primitives/scripts/installer.sh:
```diff
@@ -3,7 +3,7 @@
 export DEBIAN_FRONTEND=noninteractive
 sudo apt update -y
-sudo apt install -y curl
+sudo apt install curl gnupg openssl net-tools -y
```
are you sure you need all of there packages ?

DaMandal0rian replied:
Yes, i need to update to latest curl (used to get external IP and install docker) gnupg (update) for apt keyring, latest openssl and net-tools to check ports with netstat.

vedhavyas reviewed (commented):
Some questions and nits. Over all make sense

DaMandal0rian committed:
Changes to gemini-network configuration

- identity injection
- minor fixes
- remove alice
- add dsn-reserved-nodes / dsn-bootstrap-nodes

Major changes

- scale network rpcs and domain nodes
- add domains bootstrap node
- configuration changes to farmer and domains
- minor fixes
- dns changes for domains (add cname record)

add ingress port evm boostrap node

- minor changes (disable EIP/Nat Gateway)

Fixes

- remove NAT gateway dependency
- fix null provisioners typo

prevent terraform resource drift with AMIs

PR review fixes

+ change instance type
+ changes to dns for evm domains

Domain operator bootstrap node config

+ change instance_count variable to map

DaMandal0rian committed:
fix domains configuration

- allow only one domain operator as validator
- use keystore with main domain operator
